simd/src/Kokkos_SIMD_Common.hpp

```diff
@@ -46,8 +46,10 @@ inline constexpr simd_flags<simd_alignment_vector_aligned> simd_flag_aligned{};
 using element_aligned_tag = simd_flags<>;
 using vector_aligned_tag  = simd_flags<simd_alignment_vector_aligned>;
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
 // class template declarations for const_where_expression and where_expression
 
+KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()
```

Comment:
I'm not familiar with this, but it sounds like it disables deprecated warnings. Is that what we want?

Reply:
It let's us use deprecations without warnings in the guarded code.

Reply:
Thanks 👍 Just making sure that's what we're aiming to to here and it's not accidentally left over from development.